Leanplum-SDK/Classes/MessageTemplates/LPMessageTemplateProtocol.h

```diff
@@ -6,19 +6,17 @@
 //  Copyright © 2020 Leanplum. All rights reserved.
 //
 
-#import <Foundation/Foundation.h>
 #import "Leanplum.h"
 #import "LPMessageTemplateConstants.h"
+#import "LPApplication+Extensions.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
-@interface LPBaseMessageTemplate : NSObject
+@protocol LPMessageTemplateProtocol
 
-@property  (nonatomic, strong) NSMutableArray *contexts;
+@property (nonatomic, strong) LPActionContext *context;
```

Comment:
Maybe we can remove the context from the protocol. Because we are not really using this one.
You are using the context define in the view controller or not using the context at all (in cases where we are using alert view controller)

Reply:
yeah, there are some places not used at all, i left it only because of push notifications IAM, as soon as we move it we can remove the context from protocol

Reply:
So, we can either remove it from the protocol or set it as an optional in the protocol and only use it in templates that we need it

Reply:
updated